src/everest/config/everest_config.py

```diff
@@ -368,6 +422,30 @@ def validate_install_templates_unique_output_files(
         )
         return install_templates
 
+    @model_validator(mode="before")
+    @classmethod
+    def validate_no_data_file(cls, values: dict[str, Any]) -> dict[str, Any]:
+        if "model" in values and "data_file" in values["model"]:
```

Comment:
Should probably be on `ModelConfig`:
´´´python
    @model_validator(mode="before")
    @classmethod
    def deprecate_key(cls, values: Any) -> Any:
        if "data_file" in values:
            raise ...
```

though we might want cross validation between `eclbase` and `model.data_file` 🤔